[Bug] Caused by: java.time.format.DateTimeParseException: Text '2000-01-01T00:00:00+08:00' could not be parsed at index 19

### Search before asking

- [X] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.


### Flink version

flink-1.18.0

### Flink CDC version

flink-cdc-3.0.0

### Database and its version

mysql 8.0.30-txsql

### Minimal reproduce step

使用cdc整库同步过程中，某个表的字段：`publish_at` timestamp NULL DEFAULT '2000-01-01 00:00:00' COMMENT '课节资源计划发布时间'，同步时报错，当设置【DEFAULT NULL】后，可以正常同步。

### What did you expect to see?

不变更【DEFAULT '2000-01-01 00:00:00'】，也能正常同步

### What did you see instead?

2023-12-29 17:07:01
org.apache.flink.runtime.JobException: Recovery is suppressed by NoRestartBackoffTimeStrategy
 at org.apache.flink.runtime.executiongraph.failover.flip1.ExecutionFailureHandler.handleFailure(ExecutionFailureHandler.java:176)
 at org.apache.flink.runtime.executiongraph.failover.flip1.ExecutionFailureHandler.getFailureHandlingResult(ExecutionFailureHandler.java:107)
 at org.apache.flink.runtime.scheduler.DefaultScheduler.recordTaskFailure(DefaultScheduler.java:285)
 at org.apache.flink.runtime.scheduler.DefaultScheduler.handleTaskFailure(DefaultScheduler.java:276)
 at org.apache.flink.runtime.scheduler.DefaultScheduler.onTaskFailed(DefaultScheduler.java:269)
 at org.apache.flink.runtime.scheduler.SchedulerBase.onTaskExecutionStateUpdate(SchedulerBase.java:764)
 at org.apache.flink.runtime.scheduler.SchedulerBase.updateTaskExecutionState(SchedulerBase.java:741)
 at org.apache.flink.runtime.scheduler.SchedulerNG.updateTaskExecutionState(SchedulerNG.java:83)
 at org.apache.flink.runtime.jobmaster.JobMaster.updateTaskExecutionState(JobMaster.java:488)
 at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 at java.lang.reflect.Method.invoke(Method.java:498)
 at org.apache.flink.runtime.rpc.pekko.PekkoRpcActor.lambda$handleRpcInvocation$1(PekkoRpcActor.java:309)
 at org.apache.flink.runtime.concurrent.ClassLoadingUtils.runWithContextClassLoader(ClassLoadingUtils.java:83)
 at org.apache.flink.runtime.rpc.pekko.PekkoRpcActor.handleRpcInvocation(PekkoRpcActor.java:307)
 at org.apache.flink.runtime.rpc.pekko.PekkoRpcActor.handleRpcMessage(PekkoRpcActor.java:222)
 at org.apache.flink.runtime.rpc.pekko.FencedPekkoRpcActor.handleRpcMessage(FencedPekkoRpcActor.java:85)
 at org.apache.flink.runtime.rpc.pekko.PekkoRpcActor.handleMessage(PekkoRpcActor.java:168)
 at org.apache.pekko.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:33)
 at org.apache.pekko.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:29)
 at scala.PartialFunction.applyOrElse(PartialFunction.scala:127)
 at scala.PartialFunction.applyOrElse$(PartialFunction.scala:126)
 at org.apache.pekko.japi.pf.UnitCaseStatement.applyOrElse(CaseStatements.scala:29)
 at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:175)
 at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:176)
 at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:176)
 at org.apache.pekko.actor.Actor.aroundReceive(Actor.scala:547)
 at org.apache.pekko.actor.Actor.aroundReceive$(Actor.scala:545)
 at org.apache.pekko.actor.AbstractActor.aroundReceive(AbstractActor.scala:229)
 at org.apache.pekko.actor.ActorCell.receiveMessage(ActorCell.scala:590)
 at org.apache.pekko.actor.ActorCell.invoke(ActorCell.scala:557)
 at org.apache.pekko.dispatch.Mailbox.processMailbox(Mailbox.scala:280)
 at org.apache.pekko.dispatch.Mailbox.run(Mailbox.scala:241)
 at org.apache.pekko.dispatch.Mailbox.exec(Mailbox.scala:253)
 at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
 at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1067)
 at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1703)
 at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:172)
Caused by: java.time.format.DateTimeParseException: Text '2000-01-01T00:00:00+08:00' could not be parsed at index 19
 at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949)
 at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851)
 at java.time.Instant.parse(Instant.java:395)
 at java.util.Optional.map(Optional.java:215)
 at com.ververica.cdc.debezium.event.DebeziumSchemaDataTypeInference.inferString(DebeziumSchemaDataTypeInference.java:145)
 at com.ververica.cdc.debezium.event.DebeziumSchemaDataTypeInference.infer(DebeziumSchemaDataTypeInference.java:77)
 at com.ververica.cdc.debezium.event.DebeziumSchemaDataTypeInference.infer(DebeziumSchemaDataTypeInference.java:56)
 at com.ververica.cdc.debezium.event.DebeziumSchemaDataTypeInference.lambda$inferStruct$0(DebeziumSchemaDataTypeInference.java:198)
 at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
 at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384)
 at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
 at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
 at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:546)
 at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
 at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:438)
 at com.ververica.cdc.debezium.event.DebeziumSchemaDataTypeInference.inferStruct(DebeziumSchemaDataTypeInference.java:199)
 at com.ververica.cdc.connectors.mysql.source.MySqlSchemaDataTypeInference.inferStruct(MySqlSchemaDataTypeInference.java:40)
 at com.ververica.cdc.debezium.event.DebeziumSchemaDataTypeInference.infer(DebeziumSchemaDataTypeInference.java:81)
 at com.ververica.cdc.debezium.event.DebeziumSchemaDataTypeInference.infer(DebeziumSchemaDataTypeInference.java:56)
 at com.ververica.cdc.debezium.event.DebeziumEventDeserializationSchema.extractDataRecord(DebeziumEventDeserializationSchema.java:143)
 at com.ververica.cdc.debezium.event.DebeziumEventDeserializationSchema.extractAfterDataRecord(DebeziumEventDeserializationSchema.java:139)
 at com.ververica.cdc.debezium.event.DebeziumEventDeserializationSchema.deserializeDataChangeRecord(DebeziumEventDeserializationSchema.java:105)
 at com.ververica.cdc.debezium.event.SourceRecordEventDeserializer.deserialize(SourceRecordEventDeserializer.java:47)
 at com.ververica.cdc.debezium.event.DebeziumEventDeserializationSchema.deserialize(DebeziumEventDeserializationSchema.java:92)
 at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter.emitElement(MySqlRecordEmitter.java:128)
 at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter.processElement(MySqlRecordEmitter.java:110)
 at com.ververica.cdc.connectors.mysql.source.reader.MySqlPipelineRecordEmitter.processElement(MySqlPipelineRecordEmitter.java:119)
 at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter.emitRecord(MySqlRecordEmitter.java:82)
 at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter.emitRecord(MySqlRecordEmitter.java:55)
 at org.apache.flink.connector.base.source.reader.SourceReaderBase.pollNext(SourceReaderBase.java:160)
 at org.apache.flink.streaming.api.operators.SourceOperator.emitNext(SourceOperator.java:419)
 at org.apache.flink.streaming.runtime.io.StreamTaskSourceInput.emitNext(StreamTaskSourceInput.java:68)
 at org.apache.flink.streaming.runtime.io.StreamOneInputProcessor.processInput(StreamOneInputProcessor.java:65)
 at org.apache.flink.streaming.runtime.tasks.StreamTask.processInput(StreamTask.java:562)
 at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:231)
 at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:858)
 at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:807)
 at org.apache.flink.runtime.taskmanager.Task.runWithSystemExitMonitoring(Task.java:953)
 at org.apache.flink.runtime.taskmanager.Task.restoreAndInvoke(Task.java:932)
 at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:746)
 at org.apache.flink.runtime.taskmanager.Task.run(Task.java:562)
 at java.lang.Thread.run(Thread.java:750)

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlFullTypesITCase.java
+++ b/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlFullTypesITCase.java
@@ -133,10 +133,10 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
                     // Because Flink's BinaryWriter force write int value for TIME(6).
                     // See BinaryWriter#write for detail.
                     64822123,
-                    TimestampData.fromMillis(1595008822000L),
-                    TimestampData.fromMillis(1595008822123L),
-                    TimestampData.fromMillis(1595008822123L, 456000),
-                    LocalZonedTimestampData.fromEpochMillis(1595008822000L, 0),
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22")),
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22.123")),
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22.123456")),
+                    LocalZonedTimestampData.fromInstant(toInstant("2020-07-17 18:00:22")),
                     null
                 };
 
@@ -148,15 +148,11 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
                     64822000,
                     64822123,
                     null,
-                    TimestampData.fromMillis(1595008822000L),
-                    TimestampData.fromMillis(1595008822123L),
-                    TimestampData.fromMillis(1595008822123L, 456000),
-                    LocalZonedTimestampData.fromEpochMillis(1595008822000L, 0),
-                    LocalZonedTimestampData.fromInstant(
-                            Timestamp.valueOf("2000-01-01 00:00:00")
-                                    .toLocalDateTime()
-                                    .atZone(ZoneId.of("UTC"))
-                                    .toInstant())
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22")),
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22.123")),
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22.123456")),
+                    LocalZonedTimestampData.fromInstant(toInstant("2020-07-17 18:00:22")),
+                    LocalZonedTimestampData.fromInstant(toInstant("2000-01-01 00:00:00"))
                 };
 
         testTimeDataTypes(
@@ -192,13 +188,12 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
                     // Because Flink's BinaryWriter force write int value for TIME(6).
                     // See BinaryWriter#write for detail.
                     64822123,
-                    TimestampData.fromMillis(1595008822000L),
-                    TimestampData.fromMillis(1595008822123L),
-                    TimestampData.fromMillis(1595008822123L, 456000),
-                    LocalZonedTimestampData.fromInstant(Instant.parse("2020-07-17T18:00:22Z")),
-                    LocalZonedTimestampData.fromInstant(Instant.parse("2020-07-17T18:00:22.123Z")),
-                    LocalZonedTimestampData.fromInstant(
-                            Instant.parse("2020-07-17T18:00:22.123456Z")),
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22")),
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22.123")),
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22.123456")),
+                    LocalZonedTimestampData.fromInstant(toInstant("2020-07-17 18:00:22")),
+                    LocalZonedTimestampData.fromInstant(toInstant("2020-07-17 18:00:22.123")),
+                    LocalZonedTimestampData.fromInstant(toInstant("2020-07-17 18:00:22.123456")),
                     null
                 };
 
@@ -210,18 +205,13 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
                     64822000,
                     64822123,
                     null,
-                    TimestampData.fromMillis(1595008822000L),
-                    TimestampData.fromMillis(1595008822123L),
-                    TimestampData.fromMillis(1595008822123L, 456000),
-                    LocalZonedTimestampData.fromInstant(Instant.parse("2020-07-17T18:00:22Z")),
-                    LocalZonedTimestampData.fromInstant(Instant.parse("2020-07-17T18:00:22.123Z")),
-                    LocalZonedTimestampData.fromInstant(
-                            Instant.parse("2020-07-17T18:00:22.123456Z")),
-                    LocalZonedTimestampData.fromInstant(
-                            Timestamp.valueOf("2000-01-01 00:00:00")
-                                    .toLocalDateTime()
-                                    .atZone(ZoneId.of("UTC"))
-                                    .toInstant())
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22")),
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22.123")),
+                    TimestampData.fromTimestamp(Timestamp.valueOf("2020-07-17 18:00:22.123456")),
+                    LocalZonedTimestampData.fromInstant(toInstant("2020-07-17 18:00:22")),
+                    LocalZonedTimestampData.fromInstant(toInstant("2020-07-17 18:00:22.123")),
+                    LocalZonedTimestampData.fromInstant(toInstant("2020-07-17 18:00:22.123456")),
+                    LocalZonedTimestampData.fromInstant(toInstant("2000-01-01 00:00:00"))
                 };
 
         testTimeDataTypes(
@@ -330,6 +320,10 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
         List<Event> streamResults = fetchResults(iterator, 1);
         RecordData streamRecord = ((DataChangeEvent) streamResults.get(0)).after();
         assertThat(recordFields(streamRecord, COMMON_TYPES)).isEqualTo(expectedStreamRecord);
+    }
+
+    private Instant toInstant(String ts) {
+        return Timestamp.valueOf(ts).toLocalDateTime().atZone(ZoneId.of("UTC")).toInstant();
     }
 
     private void testTimeDataTypes(

--- a/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlFullTypesITCase.java
+++ b/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlFullTypesITCase.java
@@ -45,6 +45,7 @@ import org.testcontainers.lifecycle.Startables;
 import java.math.BigDecimal;
 import java.sql.Connection;
 import java.sql.Statement;
+import java.sql.Timestamp;
 import java.time.Instant;
 import java.time.ZoneId;
 import java.util.Arrays;
@@ -118,6 +119,7 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
                         DataTypes.TIMESTAMP(0),
                         DataTypes.TIMESTAMP(3),
                         DataTypes.TIMESTAMP(6),
+                        DataTypes.TIMESTAMP_LTZ(0),
                         DataTypes.TIMESTAMP_LTZ(0));
 
         Object[] expectedSnapshot =
@@ -134,7 +136,8 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
                     TimestampData.fromMillis(1595008822000L),
                     TimestampData.fromMillis(1595008822123L),
                     TimestampData.fromMillis(1595008822123L, 456000),
-                    LocalZonedTimestampData.fromEpochMillis(1595008822000L, 0)
+                    LocalZonedTimestampData.fromEpochMillis(1595008822000L, 0),
+                    null
                 };
 
         Object[] expectedStreamRecord =
@@ -148,7 +151,12 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
                     TimestampData.fromMillis(1595008822000L),
                     TimestampData.fromMillis(1595008822123L),
                     TimestampData.fromMillis(1595008822123L, 456000),
-                    LocalZonedTimestampData.fromEpochMillis(1595008822000L, 0)
+                    LocalZonedTimestampData.fromEpochMillis(1595008822000L, 0),
+                    LocalZonedTimestampData.fromInstant(
+                            Timestamp.valueOf("2000-01-01 00:00:00")
+                                    .toLocalDateTime()
+                                    .atZone(ZoneId.of("UTC"))
+                                    .toInstant())
                 };
 
         testTimeDataTypes(
@@ -170,7 +178,8 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
                         DataTypes.TIMESTAMP(6),
                         DataTypes.TIMESTAMP_LTZ(0),
                         DataTypes.TIMESTAMP_LTZ(3),
-                        DataTypes.TIMESTAMP_LTZ(6));
+                        DataTypes.TIMESTAMP_LTZ(6),
+                        DataTypes.TIMESTAMP_LTZ(0));
 
         Object[] expectedSnapshot =
                 new Object[] {
@@ -189,7 +198,8 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
                     LocalZonedTimestampData.fromInstant(Instant.parse("2020-07-17T18:00:22Z")),
                     LocalZonedTimestampData.fromInstant(Instant.parse("2020-07-17T18:00:22.123Z")),
                     LocalZonedTimestampData.fromInstant(
-                            Instant.parse("2020-07-17T18:00:22.123456Z"))
+                            Instant.parse("2020-07-17T18:00:22.123456Z")),
+                    null
                 };
 
         Object[] expectedStreamRecord =
@@ -206,7 +216,12 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
                     LocalZonedTimestampData.fromInstant(Instant.parse("2020-07-17T18:00:22Z")),
                     LocalZonedTimestampData.fromInstant(Instant.parse("2020-07-17T18:00:22.123Z")),
                     LocalZonedTimestampData.fromInstant(
-                            Instant.parse("2020-07-17T18:00:22.123456Z"))
+                            Instant.parse("2020-07-17T18:00:22.123456Z")),
+                    LocalZonedTimestampData.fromInstant(
+                            Timestamp.valueOf("2000-01-01 00:00:00")
+                                    .toLocalDateTime()
+                                    .atZone(ZoneId.of("UTC"))
+                                    .toInstant())
                 };
 
         testTimeDataTypes(
@@ -340,7 +355,8 @@ public class MySqlFullTypesITCase extends MySqlSourceTestBase {
 
         try (Connection connection = database.getJdbcConnection();
                 Statement statement = connection.createStatement()) {
-            statement.execute("UPDATE time_types SET time_6_c = null WHERE id = 1;");
+            statement.execute(
+                    "UPDATE time_types SET time_6_c = null, timestamp_def_c = default WHERE id = 1;");
         }
 
         List<Event> streamResults = fetchResults(iterator, 1);

--- a/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlMetadataAccessorITCase.java
+++ b/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlMetadataAccessorITCase.java
@@ -130,6 +130,7 @@ public class MySqlMetadataAccessorITCase extends MySqlSourceTestBase {
                                             DataTypes.TIMESTAMP(0),
                                             DataTypes.TIMESTAMP(3),
                                             DataTypes.TIMESTAMP(6),
+                                            DataTypes.TIMESTAMP_LTZ(0),
                                             DataTypes.TIMESTAMP_LTZ(0)
                                         },
                                         new String[] {
@@ -142,7 +143,8 @@ public class MySqlMetadataAccessorITCase extends MySqlSourceTestBase {
                                             "datetime_c",
                                             "datetime3_c",
                                             "datetime6_c",
-                                            "timestamp_c"
+                                            "timestamp_c",
+                                            "timestamp_def_c"
                                         }))
                         .build();
         assertThat(actualSchema).isEqualTo(expectedSchema);
@@ -176,7 +178,8 @@ public class MySqlMetadataAccessorITCase extends MySqlSourceTestBase {
                                             DataTypes.TIMESTAMP(6),
                                             DataTypes.TIMESTAMP_LTZ(0),
                                             DataTypes.TIMESTAMP_LTZ(3),
-                                            DataTypes.TIMESTAMP_LTZ(6)
+                                            DataTypes.TIMESTAMP_LTZ(6),
+                                            DataTypes.TIMESTAMP_LTZ(0)
                                         },
                                         new String[] {
                                             "id",
@@ -190,7 +193,8 @@ public class MySqlMetadataAccessorITCase extends MySqlSourceTestBase {
                                             "datetime6_c",
                                             "timestamp_c",
                                             "timestamp3_c",
-                                            "timestamp6_c"
+                                            "timestamp6_c",
+                                            "timestamp_def_c"
                                         }))
                         .build();
         assertThat(actualSchema).isEqualTo(expectedSchema);

--- a/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/resources/ddl/column_type_test.sql
+++ b/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/resources/ddl/column_type_test.sql
@@ -106,6 +106,7 @@ CREATE TABLE time_types
     datetime3_c          DATETIME(3),
     datetime6_c          DATETIME(6),
     timestamp_c          TIMESTAMP NULL,
+    timestamp_def_c      TIMESTAMP NULL DEFAULT '2000-01-01 00:00:00',
     PRIMARY KEY (id)
 ) DEFAULT CHARSET=utf8;
 
@@ -119,4 +120,5 @@ VALUES (DEFAULT,
         '2020-07-17 18:00:22',
         '2020-07-17 18:00:22.123',
         '2020-07-17 18:00:22.123456',
-        '2020-07-17 18:00:22');
+        '2020-07-17 18:00:22',
+        NULL);

--- a/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/resources/ddl/column_type_test_mysql8.sql
+++ b/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/resources/ddl/column_type_test_mysql8.sql
@@ -108,6 +108,7 @@ CREATE TABLE time_types
     timestamp_c          TIMESTAMP(0),
     timestamp3_c         TIMESTAMP(3),
     timestamp6_c         TIMESTAMP(6),
+    timestamp_def_c      TIMESTAMP NULL DEFAULT '2000-01-01 00:00:00',
     PRIMARY KEY (id)
 ) DEFAULT CHARSET=utf8;
 
@@ -123,4 +124,5 @@ VALUES (DEFAULT,
         '2020-07-17 18:00:22.123456',
         '2020-07-17 18:00:22',
         '2020-07-17 18:00:22.123',
-        '2020-07-17 18:00:22.123456');
+        '2020-07-17 18:00:22.123456',
+        NULL);

--- a/flink-cdc-connect/flink-cdc-source-connectors/flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/event/DebeziumSchemaDataTypeInference.java
+++ b/flink-cdc-connect/flink-cdc-source-connectors/flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/event/DebeziumSchemaDataTypeInference.java
@@ -142,7 +142,7 @@ public class DebeziumSchemaDataTypeInference implements SchemaDataTypeInference,
         if (ZonedTimestamp.SCHEMA_NAME.equals(schema.name())) {
             int nano =
                     Optional.ofNullable((String) value)
-                            .map(Instant::parse)
+                            .map(s -> ZonedTimestamp.FORMATTER.parse(s, Instant::from))
                             .map(Instant::getNano)
                             .orElse(0);
 

